Tests/CacheAdvanceTests/CacheAdvanceTests.swift

```diff
@@ -24,6 +24,47 @@ final class CacheAdvanceTests: XCTestCase {
 
     // MARK: Behavior Tests
 
+    func test_isEmpty_returnsTrueWhenCacheIsEmpty() throws {
+        let cache = try createCache()
+
+        XCTAssertTrue(try cache.isEmpty())
+    }
+
+    func test_isEmpty_returnsFalseWhenCacheHasASingleMessage() throws {
+        let message: TestableMessage = "This is a test"
+        let cache = try createCache(messages: [message])
+        try cache.append(message: message)
+
+        XCTAssertFalse(try cache.isEmpty())
+    }
+
+    func test_isEmpty_returnsFalseWhenOpenedOnCacheThatHasASingleMessage() throws {
+        let message: TestableMessage = "This is a test"
+        let cache = try createCache()
+        try cache.append(message: message)
+
+        let sameCache = try createCache(zeroOutExistingFile: false)
+
+        XCTAssertFalse(try sameCache.isEmpty())
+    }
+
+    func test_isEmpty_returnsFalseWhenCacheThatDoesNotOverwriteIsFull() throws {
```

Comment:
I don’t think this cache is full, right?

Reply:
`try createCache(messages: [message])` creates a cache that is of the exact size to store `[message]`. So this cache is full.

I can do a separate PR to make these parameters a bit more intuitive. It confused me while I was writing this too 😅